service/models.py

```diff
@@ -124,32 +178,69 @@ class ShopcartItem(db.Model):
     app = None
 
     ##################################################
-    # Shopcart Table Schema
+    # ShopcartItems Table Schema
     ##################################################
     id = db.Column(db.Integer, primary_key=True)
     sid = db.Column(db.Integer)
     sku = db.Column(db.Integer)
     name = db.Column(db.String)
     price = db.Column(db.Float)
     amount = db.Column(db.Integer)
-    create_time = db.Column(db.DateTime)
-    update_time = db.Column(db.DateTime)
+    create_time = db.Column(db.DateTime, server_default=func.now())
+    update_time = db.Column(db.DateTime, server_default=func.now(), onupdate=func.now())
 
     ##################################################
     # INSTANCE METHODS
     ##################################################
 
+    def __repr__(self):
+        return "<ShopcartItem %r>" % (self.id)
+
+    def create(self):
+        """
+        Creates a ShopcartItem to the database
+        """
+        date_time = datetime.now()
+        self.id = None  # id must be none to generate next primary key
+        if self.create_time is None:
+            self.create_time = date_time
+        if self.update_time is None:
+            self.update_time = date_time
+        db.session.add(self)
+        db.session.commit()
+
     def serialize(self):
         """ Serializes a Shopcart into a dictionary """
+        #ensure the (optional) time fields are strings
+        if isinstance(self.create_time, datetime):
+            create_time = datetime.isoformat(self.create_time)
+        else:
+            create_time = self.create_time
+
+        if isinstance(self.update_time, datetime):
+            update_time = datetime.isoformat(self.update_time)
+        else:
+            update_time = self.create_time
+
+        if self.create_time is None and self.update_time is None:
+            return{
+                "id": self.id,
+                "sid": self.sid,
+                "sku": self.sku,
+                "name": self.name,
+                "price": self.price,
+                "amount": self.amount,
+            }
```

Comment:
Same here. I think we should include `create_time` and `update_time` but set it as `None`.

Reply:
yeah you're right I'll change it